pkg/controller/certs/certs.go

```diff
@@ -40,16 +41,23 @@ func (kp *KeyPair) ToPEM() (certPEM []byte, privPEM []byte, err error) {
 	return
 }
 
-func GenerateCA() (*KeyPair, error) {
+// GenerateCA generates a self-signed CA cert/key pair that expires in expiresIn days
+func GenerateCA(expiresIn int) (*KeyPair, time.Time, error) {
+	if expiresIn > 730 || expiresIn <= 0 {
+		return nil, time.Time{}, fmt.Errorf("invalid cert expiration")
+	}
+
+	notBefore := time.Now()
+	notAfter := notBefore.AddDate(0, 0, expiresIn)
+
 	caDetails := &x509.Certificate{
 		//TODO(Nick): figure out what to use for a SerialNumber
 		SerialNumber: big.NewInt(1653),
```

Comment:
I'm wondering if you could utilize https://github.com/kubernetes/client-go/blob/982e3f55001f505adeeafb7f559affe3c7594420/util/cert/cert.go#L68 rather than doing all this. I don't think it matters what the serial number is for self-signed certificates, but otherwise I think a random number needs to be used for security reasons.

Reply:
Thanks for finding that! My only concern is that it seems like the expiration is hardcoded - though that may not even be a big deal 😄